include/cudaq/Optimizer/Builder/Intrinsics.h

```diff
@@ -86,13 +86,40 @@ class IRBuilder : public mlir::OpBuilder {
   }
 
   cc::GlobalOp
-  genVectorOfComplexConstant(mlir::Location loc, mlir::ModuleOp module,
-                             mlir::StringRef name,
-                             const std::vector<std::complex<double>> &values);
+  genVectorOfConstants(mlir::Location loc, mlir::ModuleOp module,
+                       mlir::StringRef name,
+                       const std::vector<std::complex<double>> &values);
   cc::GlobalOp
-  genVectorOfComplexConstant(mlir::Location loc, mlir::ModuleOp module,
-                             mlir::StringRef name,
-                             const std::vector<std::complex<float>> &values);
+  genVectorOfConstants(mlir::Location loc, mlir::ModuleOp module,
+                       mlir::StringRef name,
+                       const std::vector<std::complex<float>> &values);
+
+  cc::GlobalOp genVectorOfConstants(mlir::Location loc, mlir::ModuleOp module,
```

Comment:
This looks like a perfect case for a template.

Reply:
I agree and I even tried:) That made a lot of helper code move to intrinsics.h - probably most of the cpp file - would this be ok?

Reply:
Will leave this to next PR as discussed.